File: project/hyeongmin-onboarding/src/main/java/kr/co/fastcampus/onboarding/hyeongminonboarding/domain/entity/Survey.java
package kr.co.fastcampus.onboarding.hyeongminonboarding.domain.entity;

import jakarta.persistence.*;
import kr.co.fastcampus.onboarding.hyeongminonboarding.global.entity.base.BaseTimeEntity;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.List;
import java.util.UUID;


/**
 * @author khm0813
 * - 설문조사 테이블
 */
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "survey")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Survey extends BaseTimeEntity {
    @Id
    private Long id;

    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private int version;
}

File: project/sungmin-onboarding/survey-service/src/main/kotlin/com/okdori/surveyservice/domain/Survey.kt
package com.okdori.surveyservice.domain

import com.okdori.surveyservice.domain.base.BaseTimeEntity
import io.hypersistence.utils.hibernate.id.Tsid
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

/**
 * packageName    : com.okdori.surveyservice.domain
 * fileName       : survey
 * author         : okdori
 * date           : 2025. 6. 11.
 * description    :
 */

@Table(name = "t_survey")
@Entity
class Survey: BaseTimeEntity() {
    @Id @Tsid
    @Column(name = "survey_id", columnDefinition = "CHAR(13)")
    var id: String? = null

    @Column(nullable = false, columnDefinition = "VARCHAR(200)")
    var surveyName = ""

    @Column(columnDefinition = "VARCHAR(2000)")
    var surveyDescription: String? = null

    fun updateSurvey(surveyName: String?, surveyDescription: String?) {
        surveyName?.let { this.surveyName = surveyName }
        surveyDescription?.let { this.surveyDescription = surveyDescription }
    }
}
